### DISCUSSION TITLE
How to upload a secret file?

### DISCUSSION THREAD

**Discussion by ninetynin** [community/community#53469] [Actions] [answered]
### Select Topic Area

Question

### Body

I have a session.json file to login to instagram without username and password and if anyone has access to it they can login directly too so i cant upload the file in repo directly
so does anyone have any idea on how to upload the file like env variables?

**Comment by purpleblueslime** [comment]
Use [**secrets**](https://docs.github.com/en/actions/security-guides/encrypted-secrets#creating-encrypted-secrets-for-a-repository). also never upload secret/env file(s) to github.

**Reply by ninetynin** [reply]
No its a session.json file not env variables

**Comment by airtower-luna** [comment, marked as answer]
If the file is too large for a secret, there's also the option to encrypt the file, commit the encrypted file, and store the decryption key in a secret. Then you can decrypt the file in your workflow. Mind that this requires some extra care so the secret data doesn't show up in the log. See: https://docs.github.com/en/actions/security-guides/encrypted-secrets#storing-large-secrets

**Reply by ninetynin** [reply]
yep i found something similar called git-crypt 👍
